src/core/expr/head_reduce_unary.cc

```diff
@@ -749,8 +751,87 @@ static Column compute_max(Column&& arg, const Groupby& gby) {
 }
 
 
+//------------------------------------------------------------------------------
+// nunique(A:grouped)
+//------------------------------------------------------------------------------
+
+// T is the type of the input column
+
+
+template <typename T, bool NA>
+static Column _gnunique(Column&& arg, const Groupby& gby) {
+  return Column(new CountGrouped_ColumnImpl<T,NA>(std::move(arg), gby));
```

Comment:
Are you sure you can simply reuse `CountGrouped_ColumnImpl ` for `nunique`? Consider this example
```
>>>DT = dt.Frame([1, 1, 2])
>>>DT[:, {"nunique" : dt.nunique(f[0])}, dt.by(f[0])]
```
the result is the same as for `dt.count()`:
```
   |    C0  nunique
   | int32    int64
-- + -----  -------
 0 |     1        2
 1 |     2        1
[2 rows x 2 columns]
```
I would expect it returning
```
   |    C0  nunique
   | int32    int64
-- + -----  -------
 0 |     1        1
 1 |     2        1
[2 rows x 2 columns]
```

Reply:
Thanks @oleksiyskononenko, you are absolutely right, a massive oversight on my part. I will try to fix it.